pkg/scheduler/predicates/ha.go

```diff
@@ -469,3 +474,22 @@ func isPodDesired(tc *v1alpha1.TidbCluster, component, podName string) bool {
 	}
 	return ordinals.Has(ordinal)
 }
+
+func isFailureMember(tc *v1alpha1.TidbCluster, component, podName string) bool {
+	if component == v1alpha1.PDMemberType.String() {
+		for _, fm := range tc.Status.PD.FailureMembers {
+			if fm.PodName == podName {
+				return true
```

Comment:
We also need to check the `pd.members` to check the current status, because if one Pod is recovered, it may not be removed from the failureMembers.

Reply:
I think the Pod should be treated as a failure member before it is removed from the failureMembers. This won't cause any problem.

Reply:
What about a Pod failed, set in the failureMembers, deleted by the Operator, and then recovered, but the failureMember is not updated because we only clear the failureMembers when all of the failed Pods recovered, you can check the logic [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_member_manager.go#L254).